posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -0,0 +1,654 @@
+            nullable = True
+
+        clickhouse_type = re.sub(r"\(.+\)+", "", clickhouse_type)
+
+        table_columns[column_name] = {
+            "data_type": CLICKHOUSE_DLT_MAPPING[clickhouse_type],
+            "nullable": nullable,
+        }
+
+    hogql_query = saved_query.query["query"]
+
+    destination = get_dlt_destination()
+    pipeline = dlt.pipeline(
+        pipeline_name=f"materialize_model_{model_label}",
+        destination=destination,
+        dataset_name=f"team_{team.pk}_model_{model_label}",
+    )
+    _ = await asyncio.to_thread(pipeline.run, hogql_table(hogql_query, team, saved_query.name, table_columns))
+
+    tables = get_delta_tables(pipeline)
+    key, delta_table = tables.popitem()
+    return (key, delta_table)
+
+
+@dlt.source(max_table_nesting=0)
+def hogql_table(query: str, team: Team, table_name: str, table_columns):
+    """A dlt source representing a HogQL table given by a HogQL query."""
+
+    async def get_hogql_rows():
+        response = await asyncio.to_thread(execute_hogql_query, query, team)
+
+        if not response.columns:
+            raise EmptyHogQLResponseColumnsError()
+
+        columns: list[str] = response.columns
+
+        for row in response.results:
+            yield dict(zip(columns, row))
+
+    yield dlt.resource(
+        get_hogql_rows,
+        name="hogql_table",
+        table_name=table_name,
+        table_format="delta",
+        write_disposition="merge",
+        columns=table_columns,
+    )
+
+
+def get_dlt_destination():
+    if TEST:
+        credentials = {
+            "aws_access_key_id": settings.AIRBYTE_BUCKET_KEY,
+            "aws_secret_access_key": settings.AIRBYTE_BUCKET_SECRET,
+            "endpoint_url": settings.OBJECT_STORAGE_ENDPOINT,
+            "region_name": settings.AIRBYTE_BUCKET_REGION,
+            "AWS_ALLOW_HTTP": "true",
+            "AWS_S3_ALLOW_UNSAFE_RENAME": "true",
+        }
+    else:
+        credentials = {
+            "aws_access_key_id": settings.AIRBYTE_BUCKET_KEY,
+            "aws_secret_access_key": settings.AIRBYTE_BUCKET_SECRET,
+            "region_name": settings.AIRBYTE_BUCKET_REGION,
+            "AWS_S3_ALLOW_UNSAFE_RENAME": "true",
+        }
+
+    return dlt.destinations.filesystem(
+        credentials=credentials,
+        bucket_url=settings.BUCKET_URL,  # type: ignore
+        layout="modeling/{table_name}/{load_id}.{file_id}.{ext}",
+    )
+
+
+@dataclasses.dataclass
+class BuildDagActivityInputs:
+    team_id: int
+    select: list[str] = dataclasses.field(default_factory=list)
+
+
+# Pattern matches model labels prefixed and/or suffixed by a '+' and additionally numbers.
+# For example:
+# '+my_model': "Run my_model and all of its ancestors"
+# '1+my_model': "Run my_model and its direct parents"
+# '+my_model+': "Run my_model, all of its ancestors, and all of its descendants"
+# 'my_model+': "Run my_model and all of its descendants"
+# 'my_model+1': "Run my_model and its direct children"
```

Comment:
Can this not be some input to the DAG as opposed to having to parse a name to decide what to run?

Reply:
It can, happy to make it an input. I think the parsing is more relevant if this were a CLI.